src/System.Memory/tests/Performance/Perf.ReadOnlySequence.Slice.cs

```diff
@@ -2,765 +2,338 @@
-                    {
-                        ReadOnlySequence<byte> temp = buffer.Slice(start, end);
-                        localInt ^= temp.Start.GetInteger();
-                    }
-                }
-                _volatileInt = localInt;
-            }
+            public Array(): base(ReadOnlySequenceFactory<char>.ArrayFactory) { }
         }
 
-        #endregion
-
-        #region Byte_Memory
-
-        [Benchmark(InnerIterationCount = InnerCount)]
-        [InlineData(10_000, 100)]
-        private static void Byte_Memory_Long(int bufSize, int bufOffset)
+        public class Memory : Perf_ReadOnlySequence_Slice_Char
         {
-            var manager = new CustomMemoryForTest<byte>(new byte[bufSize], bufOffset, bufSize - 2 * bufOffset);
-            var buffer = new ReadOnlySequence<byte>(manager.Memory);
-            long offset = buffer.Length / 10;
-
-            foreach (BenchmarkIteration iteration in Benchmark.Iterations)
-            {
-                int localInt = 0;
-                using (iteration.StartMeasurement())
-                {
-                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
-                    {
-                        ReadOnlySequence<byte> temp = buffer.Slice(offset);
-                        localInt ^= temp.Start.GetInteger();
-                    }
-                }
-                _volatileInt = localInt;
-            }
+            public Memory() : base(ReadOnlySequenceFactory<char>.MemoryFactory) { }
         }
 
-        [Benchmark(InnerIterationCount = InnerCount)]
-        [InlineData(10_000, 100)]
-        private static void Byte_Memory_LongLong(int bufSize, int bufOffset)
+        public class String : Perf_ReadOnlySequence_Slice_Char
         {
-            var manager = new CustomMemoryForTest<byte>(new byte[bufSize], bufOffset, bufSize - 2 * bufOffset);
-            var buffer = new ReadOnlySequence<byte>(manager.Memory);
-            long offset = buffer.Length / 20;
-            long sliceLen = buffer.Length - 2 * offset;
-
-            foreach (BenchmarkIteration iteration in Benchmark.Iterations)
-            {
-                int localInt = 0;
-                using (iteration.StartMeasurement())
-                {
-                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
-                    {
-                        ReadOnlySequence<byte> temp = buffer.Slice(offset, sliceLen);
-                        localInt ^= temp.Start.GetInteger();
-                    }
-                }
-                _volatileInt = localInt;
-            }
+            public String() : base(ReadOnlySequenceFactoryChar.StringFactory) { }
         }
 
-        [Benchmark(InnerIterationCount = InnerCount)]
-        [InlineData(10_000, 100)]
-        private static void Byte_Memory_LongPos(int bufSize, int bufOffset)
+        public class Segments___1 : Perf_ReadOnlySequence_Slice_Char
```

Comment:
fix the name?

Reply:
I use it for correct ordering if we will using test for 20 segments and so on.

Reply:
In that case, can you add a comment please? Would Segments_01 work for ordering?